src/Components/Aspire.OracleManagedDataAccess/AspireOracleManagedDataAccessExtensions.cs

```diff
@@ -0,0 +1,131 @@
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="builder"/> or <paramref name="name"/> is null.</exception>
+    /// <exception cref="ArgumentException">Thrown if mandatory <paramref name="name"/> is empty.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="OracleManagedDataAccessSettings.ConnectionString"/> is not provided.</exception>
+    public static void AddKeyedOracleManagedDataAccessConnection(this IHostApplicationBuilder builder, string name, Action<OracleManagedDataAccessSettings>? configureSettings = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        AddOracleManagedDataAccessConnection(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddOracleManagedDataAccessConnection(IHostApplicationBuilder builder, string configurationSectionName,
+        Action<OracleManagedDataAccessSettings>? configureSettings, string connectionName, object? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        OracleManagedDataAccessSettings settings = new();
+        builder.Configuration.GetSection(configurationSectionName).Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ConnectionString = connectionString;
+        }
+
+        configureSettings?.Invoke(settings);
+
+        builder.RegisterOracleManagedDataAccessServices(settings, configurationSectionName, connectionName, serviceKey);
+
+        if (settings.HealthChecks)
+        {
+            builder.TryAddHealthCheck(new HealthCheckRegistration(
+                serviceKey is null ? "OracleManagedDataAccess" : $"OracleManagedDataAccess_{connectionName}",
+                sp => new OracleHealthCheck(new OracleHealthCheckOptions()
+                {
+                    Connection = serviceKey is null
+                        ? sp.GetRequiredService<OracleConnection>()
+                        : sp.GetRequiredKeyedService<OracleConnection>(serviceKey)
+                }),
+                failureStatus: default,
+                tags: default,
+                timeout: default));
+        }
+    }
+
+    private static void RegisterOracleManagedDataAccessServices(this IHostApplicationBuilder builder, OracleManagedDataAccessSettings settings, string configurationSectionName, string connectionName, object? serviceKey)
+    {
+        if (serviceKey is null)
+        {
+            // delay validating the ConnectionString until the DataSource is requested. This ensures an exception doesn't happen until a Logger is established.
+            builder.Services.AddTransient((serviceProvider) =>
+            {
+                ValidateConnection();
+
+                return new OracleConnection(settings.ConnectionString);
```

Comment:
Oracle doesn't have a data source?

Reply:
No. I intend to update this code as soon as they implement DbDataSource.

Reply:
FYI 

oracle/dotnet-db-samples#329